### PR TITLE
mysql(ticdc): add mysql backend performance metrics observer

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve?

Issue Number: ref #8045

task-1

### What is changed and how it works?

- Add a backend metrics observer.
- Fetch `tidb_conn_idle_duration`, `tidb_conn_count`, `tidb_query_duration`, `tidb_txn_duration` metrics from TiDB SQL based diagnosis.
- Unit tests will be added in another PR, in case of this PR is too large.

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: pkg/sink/observer/observer.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	sinkURIStr string,
+	replCfg *config.ReplicaConfig,
+) (Observer, error) {
+	sinkURI, err := config.GetSinkURIAndAdjustConfigWithSinkURI(sinkURIStr, replCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	schema := strings.ToLower(sinkURI.Scheme)
+	switch schema {
```
Comment (CharlesCheung96) [inline pkg/sink/observer/observer.go:43]:
```suggestion
	if !sink.IsMySQLCompatibleScheme(scheme) {
	    return NewDummyObserver(), nil
	}
```

File: pkg/sink/observer/observer.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+	"github.com/pingcap/tiflow/cdc/contextutil"
+	"github.com/pingcap/tiflow/pkg/config"
+	"github.com/pingcap/tiflow/pkg/sink"
+	pmysql "github.com/pingcap/tiflow/pkg/sink/mysql"
+)
+
+// Observer defines an interface of downstream performance observer.
+type Observer interface {
+	Tick(ctx context.Context) error
```
Comment (CharlesCheung96) [inline pkg/sink/observer/observer.go:28]: Perhaps `ObserveOnce` is a more semantically explicit method name?

Comment (amyangfei) [reply pkg/sink/observer/observer.go:30]: Still keep the `Tick` API, and add comments to this API about what it does.

**Review by CharlesCheung96 — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

File: cdc/owner/changefeed.go [hunk 3]
```diff
@@ -403,6 +416,13 @@ func (c *changefeed) tick(ctx cdcContext.Context, captures map[model.CaptureID]*
 		}
 	})
 
+	if time.Since(c.observerLastTick) > downstreamObserverTickDuration {
+		if err := c.downstreamObserver.Tick(ctx); err != nil {
```
[the remark this answers is not in the document]
Comment (amyangfei) [reply cdc/owner/changefeed.go]: Good point, add timeout and run observer tick in independent goroutine.

**Pull request merged**